src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs

```diff
@@ -630,21 +627,49 @@ private async Task<bool> EnsureConnectionStateAsync(HttpConnectionContext connec
                 {
                     // Set the request trace identifier to the current http request handling the poll
                     existing.TraceIdentifier = context.TraceIdentifier;
-                    existing.User = context.User;
+
+                    // Don't copy the identity if it's a windows identity
+                    // We specifically clone the identity on first poll if it's a windows identity
+                    // There would be a race here if we swapped the new User and the application tried to access the User
```

Comment:
To be fair, the race exists no matter what. It's `Dispose`ing the principal that caused the big problem.

Reply:
And it's the race that makes me want to stop updating the User altogether, but that would be a breaking change, so it would have to wait until 3.0 at least

Reply:
Oh right, it was dispose 😆